Libraries/LibCore/IODevice.cpp

```diff
@@ -302,6 +302,31 @@ int IODevice::printf(const char* format, ...)
     return ret;
 }
 
+Result<String, IODevice::GetLineError> IODevice::get_line(FILE* fp, bool should_chomp)
+{
+    size_t size = 0;
+    char* line = nullptr;
+    auto line_length = getline(&line, &size, fp);
+    // getline() returns -1 and sets errno=0 on EOF.
```

Comment:
You're actually supposed to check `feof()` to detect EOF in case it returns -1.

Reply:
There we go, already living up to the "supposed to" in the commit message lol
what happens if errno==0 and it returns -1, but `!feof()`?

Reply:
If stdio functions that work on a `FILE` fail, they don't typically even set errno, they set the internal error you can retrieve with `ferror()` instead.

`getline()` appears to be documented to set errno instead, which our implementation doesn't seem to do:
https://github.com/SerenityOS/serenity/blob/952c0dc2a02e92ba6380cba7fde8e4f9f00af257/Libraries/LibC/stdio.cpp#L675-L683
So you'll get some garbage in errno. If you're lucky, this will be the errno of the last syscall stdio performed, which *should* generally match the last error, but we should not rely on that.

Reply:
Noted. I'll just ignore errno in that case then.
guess I'm also fixing everything else that is using `getline()` (wrong...)